app/models/review_metric.rb

```diff
@@ -0,0 +1,134 @@
+        type = "ReviewResponseMap"
+        answers = Answer.joins("join responses on responses.id = answers.response_id")
+                       .joins("join response_maps on responses.map_id = response_maps.id")
+                       .where("response_maps.reviewed_object_id = ? and response_maps.reviewer_id = ? and response_maps.type = ? and responses.is_submitted = 1",assignment_id, reviewer_id,type)
+                       .select("answers.comments, answers.response_id, responses.round, response_maps.reviewee_id, responses.is_submitted").order("answers.response_id")
+        suggestive_words = Set.new(["should", "recommend", "suggest", "advise", "try","can", "maybe", "could", "may", "would", "should've","could've", "would've","plan"])
+        offensive_words = Set.new(["lame", "stupid", "dumb", "idiot", "horrendous","horrible","non-sense", "crap","piss","pissed","bullshit","bloody","hell","stupidity"])
+        problem_words = Set.new(["wrong", "error", "problem", "issue", "problems", "errors", "issues","problematic","fix","obstacle","dispute","not","none","didn't","cannot","can't","couldn't","shouldn't"])
+        current_response_id = nil
+        response_level_comments = Hash.new()
+        metrics = Hash.new()
+        metrics_per_reviewee = Hash.new()
+        response_reviewee_map = Hash.new()
+        diff_word_count = Hash.new()
+        complete_sentences = Hash.new(0)
+        answers.each do |ans|
+            # puts ans.comments
+            comment = ans.comments
+            response_reviewee_map[ans.response_id] = ans.reviewee_id
+            if current_response_id.nil? or current_response_id != ans.response_id
+                current_response_id = ans.response_id
+                response_level_comments[current_response_id] = comment
+            else
+                response_level_comments[current_response_id] = response_level_comments[current_response_id] + comment
+            end
+        end
+        denom = response_level_comments.length
+        response_level_comments.each_pair do |key, value|
+            word_counter = 0   
+    	    offensive_metric = 0
+            suggestive_metric = 0
+            problem_metric = 0
+            is_offensive_term = false
+            is_suggestion = false
+            is_problem = false
+            value.scan(/[\w']+/).each do |word|
+                word_counter = word_counter + 1
+    		    if offensive_words.include? word
+                    is_offensive_term = true
+                end
+                if suggestive_words.include? word
+                    is_suggestion = true
+                end
+                if problem_words.include? word
+                    is_problem = true
+                end
+            end
+
+            #diff_word_count = response_level_comments[current_response_id].scan(/[\w']+/).uniq.count
+            if ReviewMetric.exists?(response_id: key)
+                obj = ReviewMetric.find_by(response_id: key)
```

Comment:
So review metrics is only calculated when the instructor visit this information ,right?